administrator/components/com_templates/src/Model/TemplateModel.php

```diff
@@ -1810,12 +1810,20 @@ public function getMediaFiles()
         $path               = Path::clean(JPATH_ROOT . '/media/templates/' . ($template->client_id === 0 ? 'site' : 'administrator') . '/' . $template->element . '/');
         $this->mediaElement = $path;
 
+        if (!is_dir($path)) {
+            // Just in case a template does not include any media
+            mkdir($path, 0777, true);
```

Comment:
```suggestion
            Folder::create($path);
```

We shouldn't be creating any folders in core with 777